### DISCUSSION TITLE
OFV for each individual when using est="posthoc"

### DISCUSSION THREAD

**Discussion by namtien0312** [nlmixrdevelopment/nlmixr#623] [Q&A]
Dear Dr. @mattfidler and `nlmixr` team,

I have a question related to deriving the `OFV` value for each individual with a `posthoc` estimation.

When using `est="posthoc"` option (similar to `MAXEVAL = 0` in `NONMEM`), `nlmixr` estimates individual parameter based on `MAP` algorithm **on each individual patient**. Thus the estimated parameters, e.g. for patient `id = 1`, when performing this process for 100 patients is identical when performing said process just only for patient `id = 1`.

But `nlmixr` has denied `MAP` for data containing just 1 individual (error below). As a result, how can I obtain the `OFV` value when executing `MAP` for just 1 patient?
```
Error in eval(substitute(expr), data, enclos = parent.frame()) : 
'posthoc' estimation is for mixed effects models, try 'dynmodel' (need more than 1 individual)
```

Thank for your help!
Tien.

**Comment by mattfidler** [comment]
Hi @namtien0312 I would simply duplicate the single individual data and then run the "posthoc" routine.

**Comment by namtien0312** [comment]
Thank Dr. @mattfidler for the suggestion.

I have implemented this strategy before, but I found that this is time-consuming. Is there any other method available, for example, when we perform `posthoc` estimation for all individuals?

Thank you once again,
Tien.
